### ISSUE TITLE
Allow configMapRef in PackageInstall values

### ISSUE THREAD

**Issue by jduepmeier** [carvel-dev/kapp-controller#691]
**Describe the problem/challenge you have**
The `values` field in the PackageInstall allows only secrets for values files. The complete secret will be hidden in diffs. But not all config values are sensitive values. It would be nice to split the values into a combination of ConfigMap and Secrets to see the values that are not sensitive.

**Describe the solution you'd like**
Allow to fetch values from config maps.
Add a field `configMapRef` to `PackageInstallValues` and fetch use defined configmaps.
https://github.com/vmware-tanzu/carvel-kapp-controller/blob/866424d03cea80e5cb35598f2ab449defe007d87/cli/vendor/github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/packaging/v1alpha1/package_install.go#L78-L81

**Anything else you would like to add:**
I'm aware that this can be security risk if sensitive values are defined in config maps. This should be mentioned in the docs.

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

**Comment by cppforlife** [comment]
> I'm aware that this can be security risk if sensitive values are defined in config maps. This should be mentioned in the docs.

yeah, this is one of the reasons why we have so far avoided adding configMapRef. it just seems like it's way too easy to invite someone to put sensitive info into a configmap especially if PackageInstall API is presented to end user via a higher level system (like a UI) which may always end up using ConfigMap.

hoping more folks can chime in with additional opinions...
